src/client.ts

```diff
@@ -236,14 +238,39 @@ export class SkynetClient {
 
     const auth = config.APIKey ? { username: "", password: config.APIKey } : undefined;
 
+    const onDownloadProgress =
+      config.onDownloadProgress &&
+      /* istanbul ignore next */
+      function (event: ProgressEvent) {
+        let progress;
+        if (event.total === 0) {
+          // Special handling for 0-byte files to avoid NaN.
+          progress = 1;
+        } else {
+          progress = event.loaded / event.total;
+        }
+
+        // Need the if-statement or TS complains.
```

Comment:
This would work right?
```
let onDownloadProgress = undefined;
if (config.onDownloadProgress) {
    onDownloadProgress = function (event: ProgressEvent) {
        const progress = event.total ? event.loaded / event.total : 1 // avoid NaN for 0-byte file
        config.onDownloadProgress(progress, event);
    }
}
```

Reply:
I couldn't get that exact snippet to work without a TS error but I got pretty close and still managed to save a bunch of lines. Thanks.